lecture_5/homework/app/services/reports/siege_report.rb

```diff
@@ -3,10 +3,26 @@
 module Reports
   class SiegeReport
     def initialize(building:)
+      @warriors = building.warriors
```

Comment:
Gdybyśmy wynieśli wyżej tamtego ifa o którym wspomniałem w ramach tego serwisu, moglibyśmy udoskonalić nasz serwis, żeby zamiast budynku przyjmował granary i warriorów, wtedy byłby brdziej uniwersalny :) I serwis nie musi wiedzieć że robi to dla budynku :D